webapp/components/integrations/components/edit_command/index.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {getCustomTeamCommands, editCommand} from 'mattermost-redux/actions/integrations';
+import {getCommands} from 'mattermost-redux/selectors/entities/integrations';
+
+import EditCommand from './edit_command.jsx';
+
+function mapStateToProps(state, ownProps) {
+    const commandId = ownProps.location.query.id;
+
+    return {
+        ...ownProps,
+        commandId,
+        commands: Object.values(getCommands(state)),
```

Comment:
The store for commands won't change, but every time anything in the redux store changes this function will run and `getCommands(state)` will return the same object but `Object.values()` will return a new array so that `oldProps.commands !== newProps.commands`. A deep compare of the arrays would show that they are equal but pure components will check if it's exactly the same array. The websocket will still be connected here if the user comes from the channel view and even if it that couldn't happen in the future this component could be used somewhere else that could have lots of other redux store changes occuring